### PR TITLE
Docs, debugDisposed for ImmutableBuffer

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/83764
Fixes https://github.com/flutter/flutter/issues/83910

Adds `debugDisposed` to `ImmutableBuffer`, and updates a test that exercises disposed to make sure it works as intended. Updates docs. Adds a missing dispose to existing test.

Ideally, something like https://github.com/flutter/flutter/issues/83274 would help us with this, although we'd need some way to deterministically be sure that GC has run and finished for that to work.

/cc @zathras

### CODE REVIEW

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -5178,6 +5178,9 @@ class Shadow {
 }
 
 /// A handle to a read-only byte buffer that is managed by the engine.
+///
+/// The create of this object is responsible for calling [dispose] when it is
```
Comment (jason-simmons) [inline lib/ui/painting.dart:5182]: typo: "create" -> "creator"

File: lib/ui/painting.dart [hunk 2]
```diff
@@ -5178,6 +5178,9 @@ class Shadow {
...
+/// no longer needed. However,
```
Comment (jason-simmons) [inline lib/ui/painting.dart:5183]: is something missing after "However"?

Comment (dnfield) [reply lib/ui/painting.dart:5195]: I decided to put this line on the [dispose] method. Deleted.

File: lib/ui/painting.dart (hunk 1 above)
Comment (dnfield) [reply lib/ui/painting.dart:5194]: Done

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**